NBitcoin/StandardScriptTemplate.cs

```diff
@@ -148,12 +148,55 @@ protected override bool CheckScriptSigCore(Script scriptSig, Op[] scriptSigOps,
 			throw new NotSupportedException();
 		}
 
+		[Obsolete("Use GenerateWitScript instead")]
 		public Script GenerateScriptSig(TaprootSignature signature)
 		{
 			if (signature == null)
 				throw new ArgumentNullException(nameof(signature));
 			return new Script(Op.GetPushOp(signature.ToBytes()));
 		}
+
+		public PayToTaprootScriptSigParameters ExtractWitScriptParameters(WitScript witScript)
+		{
+			if (witScript.PushCount != 1 && witScript.PushCount != 2)
+				return null;
+			var hasAnex = false;
+			if (witScript.PushCount == 2)
+			{
+				if (!CheckAnnex(witScript[1]))
+					return null;
+				hasAnex = true;
+			}
+			if (!TaprootSignature.TryParse(witScript[0], out var sig))
+				return null;
+			return new PayToTaprootScriptSigParameters()
+			{
+				TransactionSignature = sig,
+				Annex = hasAnex ? witScript[1] : null,
+			};
+		}
+
+		private bool CheckAnnex(byte[] annex)
+		{
+			return annex.Length > 0 && annex[0] == 0x50;
+		}
+
+		public WitScript GenerateWitScript(TaprootSignature signature)
+		{
+			return GenerateScriptSig(signature);
+		}
+
+		public WitScript GenerateWitScript(TaprootSignature signature, byte[] annex)
+		{
+			if (signature == null)
+				throw new ArgumentNullException(nameof(signature));
+			if (annex == null)
+				throw new ArgumentNullException(nameof(signature));

```

Comment:
It should be `throw new ArgumentNullException(nameof(annex));`

But more importantly, I think annex should be able to be null. I am familiar in Bitcoin but I don't even remember what should be passed as annex.

Reply:
> It should be throw new ArgumentNullException(nameof(annex));

Fixed in https://github.com/MetacoSA/NBitcoin/pull/1082/commits/a24e96ca435db10e225e8eee9e2584ca7fb9ece9.

> I am familiar in Bitcoin but I don't even remember what should be passed as annex.

The annex is reserved for future use and is not used yet. See [this](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#cite_note-5).

> But more importantly, I think annex should be able to be null.

This is a valid point. My intention was the following: If you want to generate the witness without an annex, you should call `GenerateWitScript(TaprootSignature signature)` instead. But I agree it would be more comfortable if `GenerateWitScript(TaprootSignature signature, byte[] annex)` was able to handle `NULL` annex itself, so I fixed it in https://github.com/MetacoSA/NBitcoin/pull/1082/commits/d00e4ec06e5b2a3c836dd7e3d5196513ffbda56a.